### PR TITLE
feat: normalize email on user create and update

### PR DESCRIPTION
**Pull request by ishanarya0**
normalize email on user create and update

### CODE REVIEW

File: internal/store/postgres/user_repository.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func (r UserRepository) Create(ctx context.Context, usr user.User) (user.User, e
 	createQuery, params, err := dialect.Insert(TABLE_USERS).Rows(
 		goqu.Record{
 			"name":  usr.Name,
-			"email": usr.Email,
+			"email": strings.ToLower(usr.Email),
```
Comment (mabdh) [inline internal/store/postgres/user_repository.go]: Possible to do this in service layer?
Along with the validation whether the string is in `e-mail format` or not.

File: internal/api/v1beta1/user.go [hunk 2]
```diff
@@ -392,3 +401,8 @@ func transformUserToPB(usr user.User) (shieldv1beta1.User, error) {
 		UpdatedAt: timestamppb.New(usr.UpdatedAt),
 	}, nil
 }
+
+func isValidEmail(email string) bool {
```
Comment (mabdh) [inline internal/api/v1beta1/user.go:405]: The logic seems scattered around.
I suggest to have 1 specific function to normalize the user struct. Inside it, it should include the `checking email format` and `lowercasing the email`.

Comment (ishanarya0) [reply internal/api/v1beta1/user.go:405]: Yeah its scattered. 
I think we should keep validations in handler and business logic in service layer. So invalid email checking is correctly placed in handler.

Shall we conisder normalizing email as a business logic or validation/sanitaion step?

Comment (mabdh) [reply internal/api/v1beta1/user.go:405]: Invalid email checking should also be part of the business here I believe. It is part of user domain.

File: internal/api/v1beta1/errors.go [hunk 3]
```diff
@@ -14,6 +15,7 @@ var (
 	ErrConflictRequest       = errors.New("already exist")
 	ErrRequestBodyValidation = errors.New("invalid format for field(s)")
 	ErrEmptyEmailID          = errors.New("email id is empty")
+	ErrInvalidEmailID        = errors.New("email id is invalid")
```
Comment (AkarshSatija) [inline internal/api/v1beta1/errors.go]: Where is this being used?

Comment (AkarshSatija) [reply internal/api/v1beta1/errors.go]: Looks like this error returned is different https://github.com/goto/shield/pull/38/files#diff-875939978a63c46968edee87a7dd2766b74441e5df991143afbd911c57cfae9eR94

**Comment by AkarshSatija** [comment]
Can we add tests for the case?

**Review by mabdh — APPROVED** [review]

**Pull request merged**